### ISSUE TITLE
Wall switch with software switch

### ISSUE THREAD

**Issue by sweebee** [pimatic/pimatic#469]
I have a hardware switch and a software switch in pimatic

Hardware switch on the wall: sends signals to pimatic

if trigger wall switch is opened or softwareswitch-on is pressed then turn lights on

if trigger wall switch is closed or softwareswitch-off is pressed then turn lights off

but what happens:

when the hardware switch state is closed and i turn the lights on with the software switch, i can't switch off the lights with the hardware switch because its state is closed.

Is there a way to solve this?

this happens also when i turn off the lights with the hardware switch (then the state is closed) and i turn on 1 light in pimamatic I can't turn off that one light with the hardware switch.

Don't know what piratic does but i think:

if contact is closed in pimamatic and gets an closed signal it does nothing with it.

It should work if pimamatic overwrites the closed signal with a new closed signal like software buttons do.

**Comment by koffienl** [comment]
What brand and type of hardware switch are you using ?

**Comment by sweebee** [comment]
I'm using the KAKU AWST-8802

config:

{
      "id": "schakelaar1",
      "name": "Schakelaar 1",
      "class": "HomeduinoRFContactSensor",
      "protocols": [
        {
          "name": "contact1",
          "options": {
            "id": 14808786,
            "unit": 10
          }
        }
      ]
 
    }

**Comment by ako82** [comment]
Hey,

sorry,  I can't help you at the moment, but maybe you can help me. I just bought an Intertechno ITW-852 yesterday. I guess it is the same device you are dealing with at the moment. I wanted to integrate it in my system combination of pilight / pimatic. I already have figured out the unit and id code. But now... how can I get pimatic or pilight to do what I want when the on or off button is pressed.
I thought about something like: when I press on turn on the light in the kitchen. When I press off turn off all devices.
So basically how can I get pimatic or pilight to receive the on or off command of that device and use it in rules? I have a 433MHZ Receiver connected of course. Do I need to use homeduino for it, or can it be done with just the Pi and pilight / pimatic?

**Comment by sweebee** [comment]
I'm not using pilight, i use the homeduino plugin to add switches directly to pimatic.

then you can add this to the config.json

{
"id": "switch",
"name": "Switch",
"class": "HomeduinoRFContactSensor",
"protocols": [
{
"name": "contact1",
"options": {
"id": 14808786,
"unit": 10
}
}
]
}

and rule: if trigger: switch is opened then turn kitchenlight on

**Comment by leader21** [comment]
> Do I need to use homeduino for it, or can it be done with just the Pi and pilight / pimatic?

i would suggest not to hijack issues. but to answer quickly - you can use either the pilight plugin which automatically imports all devices that are configured in pilight or you are using pimatic with homeduino. then you do not need pilight anymore but have to configure your devices to the config.json of pimatic yourself. homeduino is surely the better way to go. if you need more information or any help please open a new issue.

**Comment by sweebee** [comment]
I fixed my own issue, i didn't know i could assign a hardware switch to a digital switch (i assigned it to a sensor). Now i can turn the hardware switch of in pimatic.
